src/main/java/com/rultor/profiles/ProfileDeprecationAware.java

```diff
@@ -0,0 +1,151 @@
+import com.jcabi.xml.XML;
+import com.rultor.spi.Profile;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Decorator allowing to add a deprecation message if and only if the
+ * {@link Profile} is deprecated.
+ * @author Nicolas Filotto (nicolas.filotto@gmail.com)
+ * @version $Id$
+ * @since 1.62
+ */
+public final class ProfileDeprecationAware implements Profile {
+
+    /**
+     * The underlying profile.
+     */
+    private final transient Profile profile;
+
+    /**
+     * Constructs a {@code ProfileDeprecationAware} with the specified
+     * underlying {@code Profile}.
+     * @param prof The underlying profile
+     */
+    public ProfileDeprecationAware(final Profile prof) {
+        this.profile = prof;
+    }
+
+    @Override
+    public String name() {
+        return this.profile.name();
+    }
+
+    @Override
+    public XML read() throws IOException {
+        final XML xml = this.profile.read();
+        this.check(xml);
+        return xml;
+    }
+
+    @Override
+    public Map<String, InputStream> assets() throws IOException {
+        return this.profile.assets();
+    }
+
+    /**
+     * Checks if the profile is deprecated and if so log the deprecation notice.
+     * @throws IOException if it fails while getting the XML format of the
+     *  profile
+     */
+    public void check() throws IOException {
+        this.check(this.profile.read());
+    }
+
+    /**
+     * Indicates whether the profile is deprecated or not.
+     * @return True if the profile is deprecated, false otherwise
+     * @throws IOException if it fails while getting the XML format of the
+     *  profile
+     */
+    public boolean deprecated() throws IOException {
+        return ProfileDeprecationAware.deprecated(this.profile.read());
+    }
+
+    /**
+     * Checks if the profile is deprecated and if so log the deprecation notice.
+     * @param prof The XML representation of the profile to check
+     */
+    private void check(final XML prof) {
+        if (ProfileDeprecationAware.deprecated(prof)) {
+            Logger.warn(this, "#### Deprecation Notice ####");
+            Logger.warn(
+                this,
+                "You are using the Rultor default Docker image in your build."
+            );
+            Logger.warn(this, "The Rultor has to:");
+            Logger.warn(
+                this,
+                "1. Provide the sudo package/command and not stop doing so "
+            );
+            Logger.warn(
+                this,
+                "whenever a change to the Dockerfile is made, even if Rultor "
+            );
+            Logger.warn(this, "itself does not need the sudo command.");
+            Logger.warn(
+                this,
+                "2. Not install any gems to the global scope that interfere "
+            );
+            Logger.warn(this, "with pdd or est");
+            Logger.warn(this, "########################");
```

Comment:
@mkordas having it in multiline is interesting for several reasons:

1. It is easier to see it in the middle of a long log file. Knowing that @original-brownbear wants this notice in several places in the log file, I guess that it means that the visibility of this notice is a key point of this ticket.
2. It allows to keep the application homogenous because in this other [task](https://github.com/yegor256/rultor/pull/1045/commits/a755cad285bf8c9897e5ddc9b7a53cd6e3e5e8fc#diff-092b503d69b9086c227ba3510a84c32dR118) the warning is printed in several lines so it makes sense to do the same here with the exact same content.

I understand your argument but are you sure that rultor is a multithreaded application (especially on start and end)?